SdarotAPI/SdarotDriver.cs

```diff
@@ -58,48 +71,32 @@ public async Task<bool> Login(string username, string password)
 
     public async Task<IEnumerable<SeriesInformation>> SearchSeries(string searchQuery)
     {
-        var searchHtml = await _httpClient.GetStringAsync($"{Constants.SdarotUrls.SearchUrl}{searchQuery}");
-        var doc = new HtmlDocument();
-        doc.LoadHtml(searchHtml);
+        var showsJson = await _httpClient.GetStringAsync($"{Constants.SdarotUrls.AllShowsUrl}");
+        var shows = JsonSerializer.Deserialize<List<SeriesInformation>>(showsJson);
+
+        var relevantShows = shows?.Where(x =>
+            x.SeriesNameHe.Contains(searchQuery, StringComparison.CurrentCultureIgnoreCase) ||
+            x.SeriesNameEn.Contains(searchQuery, StringComparison.CurrentCultureIgnoreCase)).ToList();
 
-        var seriesNameElement = doc.DocumentNode.SelectSingleNode(Constants.XPathSelectors.SeriesPageSeriesName);
+        // If none, turn to empty array.
+        relevantShows ??= new List<SeriesInformation>();
 
         // In case there is only one result
-        if (seriesNameElement is not null)
+        if (relevantShows.Count == 1)
         {
-            var seriesName = seriesNameElement.InnerText.Trim(new[] { ' ', '/' });
+            var currShow = relevantShows.First();
 
-            var imageUrlElement = doc.DocumentNode.SelectSingleNode(Constants.XPathSelectors.SeriesPageSeriesImage);
-            if (imageUrlElement is null)
-            {
-                throw new ElementNotFoundException(nameof(imageUrlElement));
-            }
-
-            var imageUrl = imageUrlElement.GetAttributeValue("src", "");
-
-            return new SeriesInformation(HttpUtility.HtmlDecode(seriesName), imageUrl).Yield();
+            return new SeriesInformation(currShow.SeriesNameHe, currShow.SeriesNameEn, currShow.ImageUrl).Yield();
         }
 
-        var seriesElements = doc.DocumentNode.SelectNodes(Constants.XPathSelectors.SearchPageResult);
-
         // In case there are no results
-        if (seriesElements is null || seriesElements.Count == 0)
+        if (!relevantShows.Any())
         {
             return Enumerable.Empty<SeriesInformation>();
         }
 
         // In case there are more than one result
```

Comment:
Seems like the code will do the exact same thing without all this code.

Reply:
This was a modification of the existing code. But sure, it seems to work, I removed it.